system-functionalities.md

```diff
@@ -338,14 +338,40 @@ Implementation guidance: For security, all verification checks run by the key se
 The asset owner can create and use signing keys in a manner similar to arbitrary secrets, with the additional operation of creating signatures.
 
 ### Inherited Operations
-All of the [protocols for arbitrary secrets](#operations-on-arbitrary-secrets) should be supported. The only difference is the type of key created and used. That is, calling the generation functionality should allow the asset owner to create one of the following key types:
+All of the [protocols for arbitrary secrets](#operations-on-arbitrary-secrets) should be supported. The only significant difference is the type of key created and used; minor protocol modifications are described below. That is, calling the generation functionality should allow the asset owner to create one of the following key types:
 - ECDSA on secp256 curve; or
 - Ed25519 (i.e., EdDSA on edwards25519).
 
-Signing keys have an additional supported operation, namely, the creation of a signature.
+Signing keys have an additional supported operation, namely, the creation of a signature. They also have an additional public component that can be derived from a generated key.
 
 Implementation guidance: We pick these ECDSA/EdDSA parameters in order to provide functionality compatible with EVM-based blockchains. However, we expect to support multiple blockchains and signing primitives in the future.
 
+### Generating and Storing a Signing Key
+Both key generation protocols need a modification to return the public component of the generated key to the calling application.
+
+In [local secret generation with remote backup](#local-secret-generation-with-remote-encrypted-backup):
+- In step [3.iii](#localgen-client-generate), the generate protocol returns the `arbitrary_key` and its public component `public_key`.
```

Comment:
the naming `arbitrary_key` is probably not so great at this point. Not sure what a better name would be, maybe just plain old `key`?

Reply:
Yeah. I can rename it `key` and/or make a note in #143 to improve the naming for the generic case.

Reply:
I would rather save this for #143 since it will be a bigger overhaul, but I can do it now if you'd prefer.

Reply:
Add it to #143.